evalml/data_checks/null_data_check.py

```diff
@@ -15,21 +15,38 @@ class NullDataCheck(DataCheck):
     Args:
         pct_null_col_threshold(float): If the percentage of NaN values in an input feature exceeds this amount,
             that column will be considered highly-null. Defaults to 0.95.
+        pct_moderately_null_col_threshold(float): If the percentage of NaN values in an input feature exceeds this amount
+             but is less than the percentage specified in pct_null_col_threshold, that column will be considered
+             moderately-null. Defaults to 0.20.
         pct_null_row_threshold(float): If the percentage of NaN values in an input row exceeds this amount,
             that row will be considered highly-null. Defaults to 0.95.
     """
 
-    def __init__(self, pct_null_col_threshold=0.95, pct_null_row_threshold=0.95):
+    def __init__(
+        self,
+        pct_null_col_threshold=0.95,
+        pct_moderately_null_col_threshold=0.20,
+        pct_null_row_threshold=0.95,
+    ):
         if pct_null_col_threshold < 0 or pct_null_col_threshold > 1:
             raise ValueError(
-                "pct null column threshold must be a float between 0 and 1, inclusive."
+                "`pct_null_col_threshold` must be a float between 0 and 1, inclusive."
+            )
+        if (
+            pct_moderately_null_col_threshold < 0
+            or pct_moderately_null_col_threshold > 1
+            or pct_moderately_null_col_threshold > pct_null_col_threshold
+        ):
+            raise ValueError(
+                "`pct_moderately_null_col_threshold` must be a float between 0 and 1, inclusive, and must be less than or equal to `pct_null_col_threshold`."
```

Comment:
I made `pct_moderately_null_col_threshold` less than or equal to `pct_null_col_threshold` instead of just less than so if a user wanted to set `pct_null_col_threshold` to 0.0 so that any column with null values returned a highly null warning, they can do so by setting `pct_moderately_null_col_threshold` as 0.0.

Reply:
Is it at all useful to return something like "0 < pct_moderately_null_col_threshold < pct_null_col_threshold < 1"

Reply:
I like it!